structlog/stdlib.py

```diff
@@ -149,6 +198,31 @@ def __call__(self, *args):
         return logging.getLogger(name)
 
 
+class StdlibFormatEventRenderer(object):
+    """
+    Applies stdlib-like string formatting to the `event` key with the arguments
+    in the `positional_args` key. This is populated by
+    `structlog.stdlib.BoundLogger` or can be manually set.
+
+    `positional_args` can be any iterable, but a dictionary as the single
+    element of the tuple is used instead of the tuple, to mantain compatibility
+    with the undocumented feature of stdlib logging.
+
+    """
+    def __init__(self, strip_positional_args=False):
+        self.strip_positional_args = strip_positional_args
+
+    def __call__(self, _, __, event_dict):
+        args = event_dict.get('positional_args')
+        if args:
+            args = tuple(args)
+            if len(args) == 1 and isinstance(args[0], dict) and args[0]:
```

Comment:
yes.

Hynek Schlawack <notifications@github.com> schreef op 15 januari 2015 09:02:29 CET:
>> +    in the `positional_args` key. This is populated by
>> +    `structlog.stdlib.BoundLogger` or can be manually set.
>> +
>> +    `positional_args` can be any iterable, but a dictionary as the
>single
>> +    element of the tuple is used instead of the tuple, to mantain
>compatibility
>> +    with the undocumented feature of stdlib logging.
>> +
>> +    """
>> +    def __init__(self, strip_positional_args=False):
>> +        self.strip_positional_args = strip_positional_args
>> +
>> +    def __call__(self, _, __, event_dict):
>> +        args = event_dict.get('positional_args')
>> +        if args:
>> +            args = tuple(args)
>> +            if len(args) == 1 and isinstance(args[0], dict) and
>args[0]:
>
>So IOW, this fragment was only there because the tests called the
>method wrong?  I figured something like that.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/hynek/structlog/pull/29/files#r22994968

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.